2019/2019-10-13-brainstorming.md

```diff
@@ -0,0 +1,122 @@
+  * Should be amazing for spectators
+* What is a subteam?
+  * Group of people collaborating on a piece of work.
+  * No more than about 6 people.
+  * Subteams should have freedom to organise themselves how they see fit.
+  * Committee should oversee operation of the subteam to ensure they are making reasonable progress.
+  * Should keep in touch with the committee on a regular basis, so that they can express their progress and any issues they're having, and we can offer ways to help.
+  * Idea: conferences - each subteam gets a chance to present their work to the rest of the kit team.
+    * Aim to build a sense of community among the whole kit team
+  * Idea: service manuals produced by development teams for use by asset management teams.
+* Ideas for subteams:
+  * Infrastructure team
+    * Provide and maintain managed hosting services to the rest of the organisation
+  * Support team
+    * Provide support to users of the kit, both inside and outside events
+    * Issues trickle down: teams -> mentors/forums -> support team -> domain experts
+  * Asset management/maintenance team
+    * Logistics, storage, tracking, maintenance (to the extent offered by service manuals)
+  * "brain" software team
+    * Robot API, operating system
+  * Development tooling team
+    * IDE, simulator, debuggers (e.g. tablet interface)
+  * Hardware strategy team
+    * Define precisely what is going to be included in the kit
+    * Can construct temporary teams to design and manufacture new boards, working in collaboration with the firmware team.
+  * Firmware team
+    * Maintain and develop the board firmwares.
+  * Robot perception and planning team
+    * Vision and motion planning software
+* Responsibilities of the committee (not an complete list):
+  * Volunteer welfare
+    * We should make it clear to volunteers that we can be contacted as individuals as well as a compound group.
+  * Volunteer onboarding
+    * This could be "outsourced" to a subteam if it's creating too much workload
+* Openness of documentation
+  * Balance needed between "open by default" and social implications of publishing something
+* Kit Team <-> Competition Team challenges
+  * Who owns team account management?
+  * Who owns forums?
+  * How to implement documentation?
+  * How to manage tasks?
+  * How to manage communication?
+* How do we review our performance?
+  * Surveys from teams
+  * Metrics e.g. how long robots move for in matches
+* How do we ascertain confidence that our software and hardware is stable? Two aspects:
+  * Making sure it is sufficiently free of bugs:
+    * Use Smallpeice summer school as a testbed?
+    * Partial rollout
+  * Ensuring that we provide a stable interface to the teams and avoid changing it while users are using it unless necessary.
+* Inventory
+  * There are a number of techncal and user experiences issues with this
+  * Ultimately, what is the inventory needed for?
+    * Knowing which teams have kit
```

Comment:
It's important to remember the inventory tracks more than just kit.

Reply:
Given that the majority of the items in the inventory are kit, what else are you referring to?

We were discussing the purpose of the inventory and think that any improvements made should be in line with that purpose. We will delegate this to our Asset sub-team once it has been formed.